### ISSUE TITLE
CUDA failed with error invalid argument

### ISSUE THREAD

**Issue by dodysw2** [guillaumekln/faster-whisper#75]
#73 introduces the following error specifically on my second GPU (cuda:1). It works fine on cuda:0.
```
File "/home/dody/sub/venv/lib/python3.10/site-packages/faster_whisper/transcribe.py", line 289, in generate_segments
    result, avg_log_prob, temperature = self.generate_with_fallback(
  File "/home/dody/sub/venv/lib/python3.10/site-packages/faster_whisper/transcribe.py", line 464, in generate_with_fallback
    result = self.model.generate(
RuntimeError: CUDA failed with error invalid argument
```

**Comment by guillaumekln** [comment]
Thank you for reporting the issue.

Can you try updating the package `ctranslate2` to `ctranslate2>=3.10.1`?

**Comment by dodysw3** [comment]
Still got the same error message. To be sure I echoed the version and it was at this:  `ctranslate2.version.__version__: 3.10.1`

For your reference, this is the relevant code: 
```python
...
    # if cuda:1 -> device=cuda, device_index=1
    device = args.device
    device_index = 0
    if device.startswith('cuda:'):
        device, device_index = device.split(':')
        device_index = int(device_index)

    # ensure latest version
    import ctranslate2
    print('ctranslate2.version.__version__:', ctranslate2.version.__version__)

    # Run on GPU with FP16
    model = faster_whisper.WhisperModel(model_path, device=device, device_index=device_index, compute_type="float16", cpu_threads=args.threads)

    writer = get_writer(args.output_format, args.output_dir)
    for audio_path in args.audio:
        segments, info = model.transcribe(
            audio_path, language=args.language, task=args.task, beam_size=args.beam_size, best_of=args.best_of, patience=args.patience, length_penalty=args.length_penalty,
            temperature=args.temperature, compression_ratio_threshold=args.compression_ratio_threshold, log_prob_threshold=args.logprob_threshold,
            no_speech_threshold=args.no_speech_threshold, condition_on_previous_text=args.condition_on_previous_text, initial_prompt=args.initial_prompt,
            word_timestamps=args.word_timestamps, prepend_punctuations=args.prepend_punctuations, append_punctuations=args.append_punctuations,
        )
...
```

**Comment by dodysw3** [comment]
By specifying `export CUDA_DEVICE_ORDER=PCI_BUS_ID` or unsetting it, I forced device index 0 to point to either one of the GPU. Can confirm that it does not matter which GPU, as long as device index is specified 1, the error occurs.

**Comment by guillaumekln** [comment]
Thanks for the details.

I think I know what's causing the error. Tomorrow I will debug on a multi-GPU instance and release a new version of CTranslate2.

**Comment by guillaumekln** [comment]
The error is fixed in `ctranslate2>=3.10.2`.

Feel free to reopen if you find something else.

**Comment by dodysw3** [comment]
Confirmed ctranslate2 3.10.2 resolves the issue.
